modules/auxiliary/scanner/ssh/ssh_enum_git_keys.rb

```diff
@@ -0,0 +1,176 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'          => 'Test SSH Github Access',
+        'Description'   => %q(
+          This module will attempt to test remote Git access using
+          (.ssh/id_* private keys). This works against GitHub and
+          GitLab by default, but can easily be extended to support
+          more server types.
+        ),
+        'License'       => MSF_LICENSE,
+        'Author'        => ['Wyatt Dahlenburg (@wdahlenb)'],
+        'Platform'      => ['linux'],
+        'SessionTypes'  => ['shell', 'meterpreter'],
+        'References'    => [['URL', 'https://help.github.com/en/articles/testing-your-ssh-connection']]
+      )
+    )
+
+    register_options(
+      [
+        OptPath.new('KEY_FILE', [false, 'Filename of a private key.', nil]),
+        OptPath.new('KEY_DIR', [false, 'Directory of several keys. Filenames will be recursivley found matching id_* (Ex: /home/user/.ssh)', nil]),
+        OptString.new('GITSERVER', [false, 'Optional parameter to specify alternate Git Server (GitHub, GitLab, etc)', 'github.com'])
```

Comment:
This probably makes more sense as a mandatory option with the default of github (which you already have done) but some sort of git server should be specified

Reply:
02bb97f should address this